src/site/topic2-cpp.rst

```diff
@@ -0,0 +1,648 @@
+* The only catch is that, unlike Java and Python, we don't get strings for free out of the box
+* We can use strings by including ``<string>``
+    * Or ``<iostream>`` since we get strings that that too
+* Also note the use of ``std`` before the string type
+
+* Just like Java, with C++ you cannot swap the types we store in our variables like we could in Python
+    * Although, there kinda' is a way to do this in C++, but we will ignore that
+
+* Again, like Java, with C++ we can declare a variable on a single like and then assign it a value later
+
+
+Constants
+---------
+
+**Python**
+
+.. code-block:: python
+    :linenos:
+
+    # Python --- Constants
+    SALES_TAX = 1.15    # This is a constant. Leave this alone!
+
+
+**Java**
+
+.. code-block:: java
+    :linenos:
+
+    // Java --- Constants
+    final double SALES_TAX = 1.15;      // Java will get mad if we mess with this
+
+
+**C++**
+
+.. code-block:: cpp
+    :linenos:
+
+    // Cpp --- Constants
+    const double SALES_TAX = 1.15;
+
+* You could also use a preprocessor definition, but using ``const`` is nicer
+* Also, don't use a preprocessor definition
+
+
+Input & Output
+==============
+
+
+**Python**
+
+.. code-block:: Python
+    :linenos:
+
+    theInput = input("gimmie something: ")
+    print(theInput)
+
+
+**Java**
+
+.. code-block:: Java
+    :linenos:
+
+    import java.io.BufferedReader;
+    import java.io.InputStreamReader;
+    import java.io.IOException;
+
+    public class SomeClass {
+        public static void main(String[] args){
+
+            // Create a Stream Reader with the standard input
+            InputStreamReader stream = new InputStreamReader(System.in);
+
+            // Give the Stream Reader to a Buffered Reader
+            BufferedReader reader = new BufferedReader(stream);
+
+            // We use the Buffered Reader to read the actual stream
+            // We use a try & catch because readLine may throw an
+            // exception that we must deal with
+            try {
+                String theLine = reader.readLine();
+                System.out.println(theLine);
+            } catch (IOException e){
+                System.out.println("Something bad happened when reading.");
+            }
+        }
+    }
+
+
+**C++**
+
+
+* Below is an example of inputting a primitive type
+
+.. code-block:: cpp
+    :linenos:
+
+    int a;
+    std::cout << "gimmie something: ";
+    std::cin >> a;
+    std::cout << a << std::endl;
+
+
+* If we want to read two things on a single line, you could do this
+
+.. code-block:: cpp
+    :linenos:
+
+    int a, b;
+    std::cout << "gimmie 2 somethings: ";
+    std::cin >> a >> b;
+    std::cout << a << " and " << b << std::endl;
+
+
+* Reading in a string is a little different, but still relatively simple
+
+.. code-block:: cpp
+    :linenos:
+    :emphasize-lines: 3
+
+    std::string a;
+    std::cout << "gimmie something: ";
+    std::getline(std::cin, a);
```

Comment:
So this isn't entirely accurate. You can still do `std::cin >> aString`, the caveat being you'll only get up to the first whitespace character. `std::getline` continues reading until it encounters a newline character.